alerta/auth/basic_ldap.py

```diff
@@ -132,17 +132,22 @@ def login():
     groups = list()
     if group_filter:
         result = ldap_connection.search_s(
-            group_base_dn or base_dn,
-            ldap.SCOPE_SUBTREE,
-            group_filter.format(username=username, email=email, userdn=userdn),
-            [current_app.config['LDAP_GROUP_NAME_ATTR']]
+            base=group_base_dn or base_dn,
+            scope=ldap.SCOPE_SUBTREE,
+            filterstr=group_filter.format(username=username, email=email, userdn=user_dn),
+            attrlist=[current_app.config['LDAP_GROUP_NAME_ATTR']] if current_app.config['LDAP_GROUP_NAME_ATTR'] else None
         )
-        for cn, group_attrs in result:
-            groups.append(group_attrs[current_app.config['LDAP_GROUP_NAME_ATTR']][0].decode('utf-8', 'ignore'))
+        for group_dn, group_attrs in result:
+            if current_app.config['LDAP_GROUP_NAME_ATTR'] in group_attrs.keys():
+                groups.append(group_attrs[current_app.config['LDAP_GROUP_NAME_ATTR']][0].decode('utf-8', 'ignore'))
+            else:
+                groups.append(group_dn)
```

Comment:
Default is to use group DN and not "cn" eg. `cn=admin_staff,ou=people,dc=planetexpress,dc=com` not just `admin_staff`